setup.md

```diff
@@ -1,8 +1,24 @@
-How to set up code for development
-- git checkout https://github.com/bpython/curtsies.git
-- cd curtsies
-- pip install -e .
-
-How to run the tests
-- pip install pyte coverage mock nose
-- nosetests .
\ No newline at end of file
+## Set up for development
+To set up a local repository to begin using curtsies as a developer, run
+
+    $ git clone https://github.com/bpython/curtsies.git
+    $ cd curtsies
+    $ pip install -e .
+
+## Running Tests
+--------
+Tests are written using the unittest framework and are run using using [pyte](https://pypi.org/project/pyte/), [coverage](https://coverage.readthedocs.io/en/coverage-5.1/),  [mock](https://pypi.org/project/mock/), and [nosetests](https://nose.readthedocs.io/en/latest/). To run all tests, do:
+
+    $ pip install pyte coverage mock nose
+    $ nosetests .
+
+Curtsies builds/pull requests are automatically tested on [Travis Ci](https://travis-ci.org/github/bpython/curtsies) following uploads, although we have ran into cases where tests pass locally but not on Travis CI. If this happens to you, be sure to document it.
```

Comment:
This is true but I'm so ashamed to admit it that I don't want it in docs, we should open an issue about the specific ones we saw.